udata/features/oembed/api.py

```diff
@@ -0,0 +1,80 @@
+
+from udata import theme
+from udata.api import api, API, output_json
+from udata.models import db, Dataset, GeoZone
+
+oembeds_parser = api.parser()
+oembeds_parser.add_argument(
+    'references', help='References of the resources to embed.',
+    location='args', required=True)
+
+
+@api.route('/oembeds/', endpoint='oembeds')
+class OEmbedsAPI(API):
+
+    @api.doc('oembeds', parser=oembeds_parser)
+    def get(self):
+        """ The returned payload is a list of OEmbed formatted responses.
+
+        See: http://oembed.com/
+
+        The `references` are composed by a keyword (`kind`) followed by
+        the `id` each of those separated by commas.
+        E.g: dataset-5369992aa3a729239d205183,territory-fr-town-75056-comptes
+
+        Only datasets and territories are supported for now.
+        """
+        args = oembeds_parser.parse_args()
+        references = args['references'].split(',')
+        result = []
+        for item_reference in references:
+            try:
+                item_kind, item_id = item_reference.split('-', 1)
+            except ValueError:
+                return api.abort(400, 'Invalid ID.')
+            if item_kind == 'dataset':
+                try:
+                    item = Dataset.objects.get(id=item_id)
+                except (db.ValidationError, Dataset.DoesNotExist):
+                    return api.abort(400, 'Unknown dataset ID.')
+                template = 'embed-dataset.html'
+            elif (item_kind == 'territory'
+                    and current_app.config.get('ACTIVATE_TERRITORIES')):
+                from udata.models import TERRITORY_DATASETS
+                try:
+                    country, town, code, kind = item_id.split('-')
+                except ValueError:
+                    return api.abort(400, 'Invalid territory ID.')
+                try:
+                    geozone = GeoZone.objects.get(code=code)
+                except GeoZone.DoesNotExist:
+                    return api.abort(400, 'Unknown territory identifier.')
+                if kind in TERRITORY_DATASETS:
+                    item = TERRITORY_DATASETS[kind](geozone)
+                else:
+                    return api.abort(400, 'Unknown kind of territory.')
+                template = 'embed-dataset.html'
+            else:
+                return api.abort(400, 'Invalid object type.')
+            width = maxwidth = 1000
+            height = maxheight = 200
+            html = theme.render(template, **{
+                'width': width,
+                'height': height,
+                'item': item,
+                'item_reference': item_reference,
+            })
+            result.append({
+                'type': 'rich',
+                'version': '1.0',
+                'html': html,
+                'width': width,
+                'height': height,
+                'maxwidth': maxwidth,
+                'maxheight': maxheight,
+            })
+        return output_json(result, 200)
```

Comment:
AFAIK, `marshal_with` needs a list of fields but we just want to return the JSON here not based on any model. Am I missing something?